avaframe/com1DFAPy/com1DFA.py

```diff
@@ -55,6 +55,177 @@
+    releaseLine = shpConv.readLine(relFiles[0], 'release1', demOri)
+    releaseLine['file'] = relFiles
+    # derive line from entrainement area polygon
+    if entFiles:
+        entLine = shpConv.readLine(entFiles, '', demOri)
+        entLine['Name'] = [os.path.splitext(os.path.basename(entFiles))[0]]
+    else:
+        entLine = None
+    # derive line from resistance area polygon
+    if resFile:
+        resLine = shpConv.readLine(resFile, '', demOri)
+        resLine['Name'] = [os.path.splitext(os.path.basename(resFile))[0]]
+    else:
+        resLine = None
+
+    flagBadName = False
+    entrainmentArea = ''
+    resistanceArea = ''
+    if flagEntRes:
+        entrainmentArea = os.path.splitext(os.path.basename(entFiles))[0]
+        resistanceArea = os.path.splitext(os.path.basename(resFile))[0]
+
+    # Counter for release area loop
+    countRel = 0
+
+    # Setup simulation dictionaries for report genereation
+    reportDictList = []
+
+    # Loop through release areas
+    for rel in relFiles:
+
+        # Set release areas and simulation name
+        relName = os.path.splitext(os.path.basename(rel))[0]
+        simName = relName
+        if '_' in relName:
+            flagBadName = True
+            log.warning('Release area scenario file name includes an underscore \
+            the suffix _AF will be added')
+            simName = relName + '_AF'
+        relDict = shpConv.SHP2Array(rel)
+        for k in range(len(relDict['d0'])):
+            if relDict['d0'][k] == 'None':
+                relDict['d0'][k] = '1.0'
+        log.info('Release area scenario: %s - perform simulations' % (relName))
+        if flagEntRes:
+            log.info('Entrainment area: %s and resistance area: %s' % (entrainmentArea, resistanceArea))
+
+        if flagEntRes:
+            # Initialise CreateSimulations cint file and set parameters
+            cuSim = [simName + '_null_dfa', simName + '_entres_dfa']
+        else:
+            # Initialise CreateSimulations cint file and set parameters
+            cuSim = [simName + '_null_dfa']
+        if flagOnlyEntrRes:
+            cuSim = [simName + '_entres_dfa']
+
+        for sim in cuSim:
+            if flagEntRes:
+                log.debug('One simulation is performed using entrainment and \
+                           one standard simulation without')
+            else:
+                log.debug('Standard simulation is performed without entrainment and resistance')
+            logName = sim + '_' + cfgGen['mu']
+            inDirPart = ''
+            if cfgGen.getboolean('initialiseParticlesFromFile'):
+                partDirName = logName
+                inDirPart = os.path.join(cfgGen['avalancheDir'], 'Outputs', 'com1DFA', 'particles', partDirName)
+            if cfgGen.getboolean('benchmarks'):
+                avaName = os.path.basename(cfgGen['avalancheDir'])
+                inDirPart = os.path.join('..', 'benchmarks', avaName, 'particles', partDirName)
```

Comment:
So if benchmark is run, particles are always used? Is this really what is wanted? Or is this for development purposes for now?